sageresearch-app-sdk/src/main/java/org/sagebionetworks/research/sageresearch/viewmodel/ScheduleRepository.kt

```diff
@@ -262,112 +210,88 @@ open class ScheduleRepository(context: Context) {
         // Which, that is okay, the s3 upload will still take place, but we won't be able to update the schedule on bridge.
         // Side effects for this will be, no study reports or finishedOn status for the task will show in the UI.
         // TODO: mdephillips 9/14/2018 message the user there will be no history?
-        val guid = scheduleTaskRunUuidMap[taskRunUuid] ?: run {
-            onError.call(Throwable("No schedule guid found for taskRunUuid $taskRunUuid, " +
+        val guid = scheduleTaskRunUuidMap[taskRunUuid]
+        return if (guid == null) {
+            Single.error(Throwable("No schedule guid found for taskRunUuid $taskRunUuid, " +
                     "are you sure you function createScheduleTaskRunUuid() before running the task?"))
-            return
+        } else {
+            toSingle(scheduleDao.activity(guid).firstOrNull(), Throwable("No schedule found in DB with guid $guid"))
         }
-
-        scheduleRepoSubscriptions.add(Observable.just(scheduleDao)
-                .subscribeOn(Schedulers.io())
-                .subscribe({ dao ->
-                    dao.activity(guid).firstOrNull()?.let {
-                        scheduleOnMain(it, onNext, onError)
-                    } ?: run {
-                        val error = Throwable("No schedule found in DB with guid $guid")
-                        scheduleOnMain(error, onError)
-                    }
-                }, {
-                    scheduleOnMain(it, onError)
-                }))
     }
 
-    /**
-     * Helper function to schedule calls on the main thread
-     */
-    @VisibleForTesting
-    protected open fun scheduleOnMain(
-            schedule: ScheduledActivityEntity,
-            onNext: Action1<ScheduledActivityEntity>,
-            onError: Action1<Throwable>) {
-
-        scheduleRepoSubscriptions.add(Observable.just(schedule)
-                .subscribeOn(AndroidSchedulers.mainThread())
-                .subscribe({
-                    onNext.call(it)
-                }, {
-                    onError.call(it)
-                }))
+    @CheckReturnValue
+    internal fun <T> toSingle(nullableValue: T?, throwableForNull: Throwable): Single<T> {
+        nullableValue?.let {
+            return Single.just(nullableValue)
+        } ?: run {
+            return Single.error(throwableForNull)
+        }
     }
 
-    /**
-     * Helper function to schedule calls on the main thread
-     */
-    @VisibleForTesting
-    protected open fun scheduleOnMain(
-            throwable: Throwable,
-            onError: Action1<Throwable>) {
-
-        scheduleRepoSubscriptions.add(Observable.just(throwable)
-                .subscribeOn(AndroidSchedulers.mainThread())
-                .subscribe({
-                    onError.call(it)
-                }, {
-                    onError.call(it)
-                }))
+    @CheckReturnValue
+    internal fun <T> toMaybe(nullableValue: T): Maybe<T> {
```

Comment:
This definitely needs some sort of function header comment.  I also feel like this shouldn't be here and should be an extension function on Maybe objects.

Reply:
Removed